src/server/array_func.c

```diff
@@ -872,8 +848,8 @@ create_subjob(job *parent, char *newjid, int *rc)
 
 	subj->ji_qs.ji_svrflags &= ~JOB_SVFLG_ArrayJob;
 	subj->ji_qs.ji_svrflags |=  JOB_SVFLG_SubJob;
-	subj->ji_modified = 1;	/* ** will likely take this out ** */
-
+	subj->ji_modified = 0;  /* to avoid db save in svr_setjobstate()*/
```

Comment:
I don't think this statement is required here since the structure subj is already initialized and will be having subj->ji_modified = 0.

Reply:
I suggest we leave the statement here with the comment to explicitly show that we are doing this to NOT save the subjob at this point of time.

Reply:
Besides we should count how many times a subjob gets saved by adding a trace in job_save() - for information purpose only.

Reply:
>Besides we should count how many times a subjob gets saved by adding a trace in job_save() - for information purpose only.

Eliminated the explicit new job db save in create_subjob() in review commit 15 https://github.com/PBSPro/pbspro/pull/590/commits/bf1558fc183bcfedea1c42d6e44c38c45870d519 Do we still need to do this?